packages/utils/src/json-schema.ts

```diff
@@ -25,7 +25,23 @@ export type JSONSchema4TypeName =
 /**
  * @see https://tools.ietf.org/html/draft-zyp-json-schema-04#section-3.5
  */
-export type JSONSchema4Type = boolean | number | string | null;
+export type JSONSchema4Type =
+  | boolean
+  | JSONSchema4Array
+  | JSONSchema4Object
```

Comment:
I don't think it is the best thing for us to add this here because `JSONSchema4Type` is used in the `enum` property where arrays and objects are disallowed (slash non-sensical).

So I think it makes sense for us to define a new type `JSONSchema4DefaultValue` which includes all of the valid default values.

Reply:
@bradzacher Updated with a separate type named `JSONSchema4TypeExtended`, which I thought could be a better name, but can change it to `JSONSchema4DefaultValue` if needed